### PR TITLE
Image registration method2 c sharp example

### PR DESCRIPTION
**Pull request by pettaa123**

### CODE REVIEW

**Comment by blowekamp** [comment]
This is absolutely terrific!

Unfortunately, none of the CI are testing CSharp. I'll add that this afternoon so we can have confidence with these great example contributions!

File: Examples/ImageRegistrationMethod2/ImageRegistrationMethod2.cs [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+        {
+            VectorDouble pos = m_Method.GetOptimizerPosition();
+            Console.WriteLine("{0:3} = {1:10.5} : [{2}, {3}]",
+                              m_Method.GetOptimizerIteration(),
+                              m_Method.GetMetricValue(),
+                              pos[0], pos[1]);
+        }
+    }
+
+    class ImageRegistrationMethod1
```
Comment (zivy) [inline Examples/ImageRegistrationMethod2/ImageRegistrationMethod2.cs]: A bit strange that the class is named ...Method1 when the example number is 2. Possibly use a generic name ImageRegistrationMethodExample or specific name ImageRegistrationMethodExample2.

File: Examples/ImageRegistrationMethod2/ImageRegistrationMethod2.cs [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+
+            R.SetInitialTransform(new TranslationTransform(fixedImage.GetDimension()));
+            R.SetInterpolator(InterpolatorEnum.sitkLinear);
+
+            IterationUpdate cmd = new IterationUpdate(R);
+            R.AddCommand(EventEnum.sitkIterationEvent, cmd);
+
+            Transform outTx = R.Execute(fixedImage, movingImage);
+
+            // System.out.println("-------");
```
Comment (zivy) [inline Examples/ImageRegistrationMethod2/ImageRegistrationMethod2.cs]: Remove commented out code and the 'using System' or un-comment these lines.

**Comment by blowekamp** [comment]
Thank you very much for the PR. Can you please rebase onto the current master so that the CSharp examples will be tested by the CI?

**Comment by blowekamp** [comment]
We are getting the following error  during CMake configuration (https://open.cdash.org/viewBuildError.php?buildid=5848495):
```
    Unknown arguments specified
  Call Stack (most recent call first):
    CMake/UseCSharp.cmake:46 (CSHARP_ADD_PROJECT)
    CMake/sitkAddTest.cmake:295 (csharp_add_executable)
    Examples/ImageRegistrationMethod2/CMakeLists.txt:21 (sitk_add_csharp_test) 
```

Please see here for proper usage of `sitk_add_csharp_test`: https://github.com/SimpleITK/SimpleITK/blob/master/Examples/ImageRegistrationMethod1/CMakeLists.txt#L21-L30

**Comment by pettaa123** [comment]
What does the workflow look like if i test the example locally? Run Cmake-gui, delete cache, create projects files with checked examples and testing, Load SuperbuildSimpleITK in VS, build SimpleITK-examples and SimpleITK-test?

**Comment by blowekamp** [comment]
I generally use the command prompts, but is doable with the GUI. You should use CMake to configure against the "SimpleITK/Superbuild"  subdirectory. And ensure that Examples are turned on. Open the Superbuild VS project and compile for  release. Close VS, then open the SimpleITK project in the "SimpleITK-build" sub-directory. This should have a "RUN_TESTS" target, which should run all the SimpleITK tests.

Alternatively after building the super build project you can open the command project and change directories into the "SimpleITK-build" sub-directory of the Superbuild project. From there you can run "ctest -C Release -R Example" to run just the Example tests.

You can find more information about building SimpleITK here:
https://simpleitk.readthedocs.io/en/master/Documentation/docs/source/building.html

**Comment by blowekamp** [comment]
I have squashed and rebased you PR. After CI we will merge it.

**Comment by blowekamp** [comment]
Merging! Thank you so much for your contribution and your patience while we got CSharp CI working.

**Pull request merged**